rasterio/_base.pxd

```diff
@@ -21,7 +21,7 @@ cdef class DatasetReader:
     cdef public object _nodatavals
     cdef public object _read
 
-    cdef void *band(self, int bidx) except NULL
+    cdef void *band(self, object bidx) except NULL
```

Comment:
@sgillies why was this changed from `int` to `object`?  Seems like down at this level, it should be an `int`...